File: 01_ProgrammingBasics/_06_NestedLoopsExercise/_04_TrainTheTrainers.java
package _01_ProgrammingBasics._06_NestedLoopsExercise;

import java.util.Scanner;

public class _04_TrainTheTrainers {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        int n = Integer.parseInt(scan.nextLine());
        String presentationName = scan.nextLine();

        // Add variables
        double totalAverage = 0;
        int presentationCount = 0;

        // Evaluate all presentations
        while (!presentationName.equals("Finish")) {
            presentationCount++;
            double presentationAverage = 0;

            for (int i = 1; i <= n; i++) {
                double grade = Double.parseDouble(scan.nextLine());
                presentationAverage += grade;
            }
            
            totalAverage += presentationAverage/n;
            
            System.out.printf("%s - %.2f.%n", presentationName, presentationAverage/n);
            presentationName = scan.nextLine();
        }
        
        double finalAverage = totalAverage / presentationCount;
        
        // Print the result
        System.out.printf("Student's final assessment is %.02f.%n", finalAverage);
        
        scan.close();
    }
}

File: 23_C#Basics/_06_NestedLoopsExercise/_04_TrainTheTrainers/_04_TrainTheTrainers.cs
using System;

public class _04_TrainTheTrainers
{
    public static void Main(string[] args)
    {
        // Read input values
        int n = int.Parse(Console.ReadLine());
        string presentationName = Console.ReadLine();

        // Add variables
        double totalAverage = 0;
        int presentationCount = 0;

        // Evaluate all presentations
        while (!presentationName.Equals("Finish"))
        {
            presentationCount++;
            double presentationAverage = 0;

            for (int i = 1; i <= n; i++)
            {
                double grade = double.Parse(Console.ReadLine());
                presentationAverage += grade;
            }

            totalAverage += presentationAverage / n;

            Console.WriteLine($"{presentationName} - {presentationAverage / n:F2}.");
            presentationName = Console.ReadLine();
        }

        double finalAverage = totalAverage / presentationCount;

        // Print the result
        Console.WriteLine($"Student's final assessment is {finalAverage:F2}.");
    }
}
